app/models/certificate.rb

```diff
@@ -1,6 +1,15 @@
+require "openssl"
+
 # Certificate store
 class Certificate < ActiveRecord::Base
   has_many :certificate_services, dependent: :destroy
 
   validates :certificate, presence: true
+  validate :valid_certificate_format
```

Comment:
Not sure if it's overengineering but could we extract this to something similar to the `EmailValidator` example at http://guides.rubyonrails.org/active_record_validations.html#custom-validators?

This way I could reuse elsewhere by just `validates :field, certificate: true`. Otherwise, how would I reuse this kind of validation?

Reply:
@vitoravelino I assumed that every certificate would go through the `certificate.rb` class in the end.

But I think in general having the `validator` separate is better design - as we might need different ones for `PKCS` certificates for some views.

Reply:
> @vitoravelino I assumed that every certificate would go through the certificate.rb class in the end.

I thought you had assumed that, so it's fine. The problem is that for simpler features we are storing directly though Pillar instead of its own model. 

> But I think in general having the validator separate is better design - as we might need different ones for PKCS certificates for some views.

That's also true.